playgame.py

```diff
@@ -102,40 +163,40 @@ def is_valid_move(self, player_move):
-        elif move == "discard":
-            return self.table.discard_card(self.current_player, player_move["card"])
-        elif move == "disclose":
+            self.disqualify_and_exit(player_move, logger)
+        move = player_move['play_type']
+        if move == 'play':
+            return self.table.play_card(self.current_player, player_move['card'])
+        elif move == 'discard':
+            return self.table.discard_card(self.current_player, player_move['card'])
+        elif move == 'disclose':
             return self.play_disclose(player_move)
 
     def play_disclose(self, player_move):
-        disclose_type = player_move["disclose_type"]
-        if disclose_type == "color":
-            return self.table.disclose_color(self.current_player, player_move["player"], player_move["color"])
-        elif disclose_type == "rank":
-            return self.table.disclose_rank(self.current_player, player_move["player"], player_move["rank"])
-
-    def disqualify_and_exit(self, bot_move):
-        logging.error("Received invalid move from player {id}".format(id = self.current_player))
-        logging.error(bot_move)
-        logging.error("Expected format for play card:")
-        logging.error("{'play_type':'play', 'card':<number>}")
-
-        logging.error("Expected format for discard card:")
-        logging.error("{'play_type':'discard', 'card':<number>}")
+        disclose_type = player_move['disclose_type']
+        if disclose_type == 'color':
+            return self.table.disclose_color(self.current_player, player_move['player'], player_move['color'])
+        elif disclose_type == 'rank':
+            return self.table.disclose_rank(self.current_player, player_move['player'], player_move['rank'])
+
+    def disqualify_and_exit(self, bot_move, logger):
+        logger.error('Received invalid move from player {id}'.format(id = self.current_player))
+        logger.error(bot_move)
+        logger.error('Expected format for play card:')
+        logger.error('{"play_type":"play", "card":<number>}')
+
+        logger.error('Expected format for discard card:')
+        logger.error('{"play_type":"discard", "card":<number>}')
         
-        logging.error("Expected format for disclose color:")
-        logging.error("{'play_type':'disclose', 'disclose_type':'color, 'color':<color>}")
-        logging.error("'color' cannot be '*' in a Variant 3 game")
+        logger.error('Expected format for disclose color:')
+        logger.error('{"play_type":"disclose", "disclose_type":"color, "color":<color>}')
+        logger.error('"color" cannot be "*" in a Variant 3 game')
         
-        logging.error("Expected format for disclose rank:")
-        logging.error("{'play_type':'disclose', 'disclose_type':'rank, 'rank':<number>}")
-        exit()
+        logger.error('Expected format for disclose rank:')
+        logger.error('{"play_type":"disclose", "disclose_type":"rank, "rank":<number>}')
+        exit(1)
```

Comment:
Why `exit` instead of [`raise`](https://docs.python.org/2/tutorial/errors.html#raising-exceptions)?

Reply:
Old plan to just stop the process. Needs to be an exception now